src/tlo/methods/contraception.py

```diff
@@ -645,24 +645,89 @@ def select_contraceptive_following_birth(self, mother_id, mother_age):
         # TODO: update with optional items (currently all considered essential)
 
-        get_items_from_pkg = self.sim.modules['HealthSystem'].get_item_codes_from_package_name
+        # ### Get item codes and number of units per case from package name
+        # get_items_from_pkg = self.sim.modules['HealthSystem'].get_item_codes_from_package_name
+
+        # _cons_codes = dict()
+        # # items for each method that requires an HSI to switch to
+        # _cons_codes['pill'] = get_items_from_pkg('Pill')
+        # _cons_codes['male_condom'] = get_items_from_pkg('Male condom')
+        # _cons_codes['other_modern'] = get_items_from_pkg('Female Condom')
+        # # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
+        # _cons_codes['IUD'] = get_items_from_pkg('IUD')
+        # _cons_codes['injections'] = get_items_from_pkg('Injectable')
+        # _cons_codes['implant'] = get_items_from_pkg('Implant')
+        # _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        # assert set(_cons_codes.keys()) == set(self.states_that_may_require_HSI_to_switch_to)
+        # # items used when initiating a modern reliable method after not using or switching from non-reliable method
+        # _cons_codes['co_initiation'] = get_items_from_pkg('Contraception initiation')
+
+        # ### Get item codes from item names and define number of units per case here
+        get_item_code = self.sim.modules['HealthSystem'].get_item_code_from_item_name
 
         _cons_codes = dict()
-        # items for each method that requires an HSI to switch to
-        _cons_codes['pill'] = get_items_from_pkg('Pill')
-        _cons_codes['male_condom'] = get_items_from_pkg('Male condom')
-        _cons_codes['other_modern'] = get_items_from_pkg('Female Condom')
-        # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
-        _cons_codes['IUD'] = get_items_from_pkg('IUD')
-        _cons_codes['injections'] = get_items_from_pkg('Injectable')
-        _cons_codes['implant'] = get_items_from_pkg('Implant')
-        _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        # # items for each method that requires an HSI to switch to
+        # TODO: @TimH Pills are alternatives, we weight their use by the probabilities (as approximation of use in 80:20
+        #  cases, for each case 0.8 and 0.2 proportion of units is used). Is that okay, or do we want to choose for each
+        #  case only one type of pills? Total usage should be approximately same.
+        # TODO: @Sakshi what are the numbers of pills in monthly packets for both following items?
+        _cons_codes['pill'] =\
+            {get_item_code("Levonorgestrel 0.0375 mg, cycle"): 28*3.75*0.2,  # progesterone-only pills used in 20% cases
```

Comment:
This will make any checks on availability require BOTH to be available. Probably better to choose one here OR chose between these probabilistically at point of dispensation in hsi rather than like this.

Reply:
@tbhallett, Good point, the alternatives are now chosen by probs, including the item Atropine that is used only in 50% cases (with no altrenative).